GraphResponse.md

```diff
@@ -7,6 +7,206 @@ Provide a decorator for the native response message to enable attaching Graph sp
+
+
+## Current Implementations
+
+### Javascript
+
+The Javascript SDK exposes the native [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) object when selecting the RAW ResponseType option.
+
+### .Net
+
+> Currently not supported
+
+### Java
+
+> Currently not supported. Should be able to use the same pattern as .Net.
+
+### PHP
+
+The PHP SDK returns a Graph model (or collection) or a GraphResponse object, depending on whether a returnType is specified. This means that the returnType information is not supplied to the GraphResponse. The response body is set as a JSON array on GraphResponse. The only work I'd suggest here is that we add an option to pass the returnType to the GraphResponse and add a lazy deserialize method so it is easy to get the Graph models.
+
+## The GraphResponse Object
+
+The GraphResponse object can wrap the native HttpResponseMessage object so that customers can use the familiar HttpResponseMessage members. 
+GraphResponse will be part of Microsoft.Graph.Core and it will expose the headers,Content and status code while also keeping reference to the Request object as below.
+
+```csharp
+
+    /// <summary>
+    /// The GraphResponse Object
+    /// </summary>
+    public class GraphResponse 
+    {
+        /// <summary>
+        /// The GraphResponse Constructor
+        /// </summary>
+        /// <param name="iBaseRequest">The Request made for the response</param>
+        /// <param name="httpResponseMessage">The response</param>
+        public GraphResponse(IBaseRequest iBaseRequest,HttpResponseMessage httpResponseMessage)
+        {
+            this.httpResponseMessage = httpResponseMessage;
+            this.BaseRequest = iBaseRequest;
+        }
+
+        private readonly HttpResponseMessage httpResponseMessage;
+        
+        /// <summary>
+        /// The Response Status code
+        /// </summary>
+        public HttpStatusCode StatusCode => httpResponseMessage.StatusCode;
+        
+        /// <summary>
+        /// The Response Content
+        /// </summary>
+        public HttpContent Content => httpResponseMessage.Content;
+        
+        /// <summary>
+        /// The Response Headers
+        /// </summary>
+        public HttpResponseHeaders HttpHeaders => httpResponseMessage.Headers;
+
+        /// <summary>
+        /// The reference to the Request
+        /// </summary>
+        public IBaseRequest BaseRequest;
+
+        /// <summary>
+        /// Get the native Response Message
+        /// </summary>
+        /// <returns></returns>
+        public HttpResponseMessage ToHttpResponseMessage()
+        {
+            return httpResponseMessage;
+        }
+    }
+
+```
+
+## Exposing the GraphResponse object from the Service Library
+
+We can create parallel *Async functions to enable the GraphResponse object functionality. We will generate requests with the following public API signatures:
+
+* `GetWithGraphResponseAsync(): : GraphResponse`
```

Comment:
I think may be `GetWithGraphResponseAsync<T>() : GraphResponse<T>`

Reply:
I'm not sure we need the "WithGraph" part.  I think GetResponseAsync<T> is sufficient.

Reply:
Resolved with both suggestions